lib/scholar/cluster/utils.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Scholar.Cluster.Utils do
+  import Nx.Defn
+  import Scholar.Shared
+
+  @spec squared_euclidean(Nx.t(), Nx.t()) :: Nx.t()
+  defn squared_euclidean(x, y, opts \\ []) do
```

Comment:
We already have this function under Scholar.Metrics.Distance

Reply:
But not with axes specification

Reply:
You can add it there instead

Reply:
I guess that we should add this option to all the other metrics to keep API consistent then, but it's an idea.

Reply:
I agree. Perhaps it's better that this is done in a separate PR first

Reply:
@msluszniak please change `Scholar.Metrics.Distance` to support the `:axes` option. :)